app/src/main/java/org/wikipedia/views/ImagePreviewDialog.kt

```diff
@@ -104,39 +103,56 @@ class ImagePreviewDialog : ExtendedBottomSheetDialogFragment(), DialogInterface.
             addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_description_in_language_title,
                     WikipediaApp.getInstance().language().getAppLanguageLocalizedName(suggestedEditsSummary.lang)),
-                    suggestedEditsSummary.description, false)
+                    SpannableString(suggestedEditsSummary.description), null)
         } else {
             addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_caption_in_language_title,
                     WikipediaApp.getInstance().language().getAppLanguageLocalizedName(suggestedEditsSummary.lang)),
-                    if (suggestedEditsSummary.pageTitle.description.isNullOrEmpty()) suggestedEditsSummary.description
-                    else suggestedEditsSummary.pageTitle.description, false)
+                    SpannableString(if (suggestedEditsSummary.pageTitle.description.isNullOrEmpty()) suggestedEditsSummary.description
+                    else suggestedEditsSummary.pageTitle.description), null)
         }
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_artist), suggestedEditsSummary.metadata!!.artist(), false)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_date), suggestedEditsSummary.metadata!!.dateTime(), false)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_source), suggestedEditsSummary.metadata!!.imageDescriptionSource(), true)
-        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_licensing), suggestedEditsSummary.metadata!!.licenseShortName(), true)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_artist), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.artistUrl()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_date), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.dateTime()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_source), StringUtil.fromHtml(suggestedEditsSummary.metadata!!.creditUrl()), null)
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_licensing), SpannableString(suggestedEditsSummary.metadata!!.licenseShortName()), getClickListenerFor(suggestedEditsSummary.metadata!!.licenseUrl()))
+        addDetailPortion(getString(R.string.suggested_edits_image_preview_dialog_more_info), SpannableString(getString(R.string.suggested_edits_image_preview_dialog_file_page_link_text)), getClickListenerFor(getString(R.string.suggested_edits_image_file_page_commons_link, suggestedEditsSummary.title)))
         detailsHolder.requestLayout()
     }
 
-    private fun addDetailPortion(titleString: String, @Nullable detail: String?, shouldAddAccentTint: Boolean) {
-        if (!detail.isNullOrEmpty()) {
+    private fun addDetailPortion(titleString: String, @Nullable detail: Spanned, linkClickListener: View.OnClickListener?) {
```

Comment:
Please don't use `@Nullable` annotation for Kotlin because you can use `?` for the type.